oak-blob-composite/src/main/java/org/apache/jackrabbit/oak/blob/composite/DelegateDataStore.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.jackrabbit.oak.blob.composite;
+
+import com.google.common.collect.Maps;
+import org.apache.jackrabbit.oak.spi.blob.DataStoreProvider;
+
+import java.util.Map;
+
+public class DelegateDataStore {
+    private static final String READ_ONLY = "readOnly";
+
+    private final DataStoreProvider ds;
+    private final Map<String, Object> config;
+
+    DelegateDataStore(final DataStoreProvider ds, final Map<String, Object> config) {
+        this.ds = ds;
+        if (null != config) {
+            this.config = config;
+        }
+        else {
+            this.config = Maps.newConcurrentMap();
+        }
+        this.config.putIfAbsent(DataStoreProvider.ROLE, ds.getRole());
+    }
+
+    public DataStoreProvider getDataStore() {
+        return ds;
+    }
+
+    public String getRole() {
+        return ds.getRole();
+    }
+
+    public Map<String, ?> getConfig() {
+        return config;
+    }
+
+    public boolean isReadOnly() {
```

Comment:
Shouldn't there be a role-specific method here rather than a specific method for a specific implementation.

Reply:
Roles were intended to be free-form text, based on feedback from @trekawek (see https://markmail.org/message/k6gl6v3rmq357tj2).

Reply:
Exactly what I am alluding to. The methods should be adjusted to not be for specific roles. the {{isReadOnly}} here is specific to the 'read-only' role and will not be relevant for other roles configured say 2 different writable delegates for different parts of the repository tree. If required the method should change to return that free-form text role.